src/python/arcor2_arserver/rpc/project.py

```diff
@@ -1127,23 +1223,27 @@ async def add_logic_item_cb(req: srpc.p.AddLogicItem.Request, ui: WsClient) -> N
     assert glob.PROJECT
     assert glob.SCENE
 
-    logic_item = common.LogicItem(req.args.start, req.args.end, req.args.condition)
-    check_logic_item(glob.PROJECT, logic_item)
+    async with ctx_write_lock(req.args.end, "ui.owner"):
```

Comment:
U těch logických propojů si nejsem úplně jistý jak to řešit. V AREditoru je totiž možné spoj táhnout jak od výstupu jedné akce na vstup jiné, tak opačně, tj. od vstupu na výstup.

Reply:
Plus se teď ještě dívám, že v `update_logic_item_cb` zamčení není.

Reply:
Nejlepší řešení co mě zatím napadlo je zamknout akci ze které táhnu a potom přes kontext manažera zamknout druhou, teď mě napadá, že si nejsem jistý, jak se daná akce vlastně posílá (jestli je vždy start startovní akcí nebo akcí od které táhnu -- zjistím a kdyžtak ještě upravím)

`update_logic_item_cb` je jedno z RPC, které jsem neuměl navodit, protože změna spoje je řešena remove > add... takových nedořešených RPC tady pár je, budu řešit nakonci, protože určitě ještě nějaké přibudou

Reply:
`start` je vždy startovní akcí, tj. výstupem nějaké akce. Ale v editoru můžu propojování začít i od `end`. Pravda, `update_logic_item_cb` se asi zatím nepoužívá.